newIDE/app/src/VariablesList/index.js

```diff
@@ -153,35 +153,28 @@ export default class VariablesList extends React.Component<Props, State> {
     );
   }
 
-  _isVariableInherited = name =>{
-    const { inheritedVariablesContainer } = this.props;
-
-    return inheritedVariablesContainer
-    ? inheritedVariablesContainer.has(name) ///<-- always false when variable is a child :(
-    : false;
-  }
-
   _renderVariableAndChildrenRows(
     name: string,
     variable: gdVariable,
     depth: number,
     index: number,
     parentVariable: ?gdVariable,
-    isParentInherited =false
+    parentType: string = '',
   ) {
     const { variablesContainer, inheritedVariablesContainer } = this.props;
     const isStructure = variable.isStructure();
 
-    const isInherited = isParentInherited || (depth === 0 && this._isVariableInherited(name))
-
-  const defaultValue =
-    inheritedVariablesContainer && isInherited && !isStructure && depth === 0 ///<- workaround for below
-      ? inheritedVariablesContainer.get(name).getString() ///<-- causes GD to crash when child variable
-      : '';
-
-      // console.log(name)
-      // console.log(inheritedVariablesContainer.get(name))
-    const variableMetadata ={isInherited,defaultValue}
+    /// We check for 3 types of variable states, when editing instance variables
+    const type = parentType.length? parentType : inheritedVariablesContainer
```

Comment:
I think it's a good idea and moving in the good direction. It gives more clarity :)